docs/modules/ROOT/pages/references/architecture/control-api-design.adoc

```diff
@@ -0,0 +1,181 @@
+* Create a new organization if I'm allowed to
+* Update an existing organization if I'm allowed to
+* Delete an existing organization if I'm allowed to
+* Invite a user to an organization if I'm allowed to send invitations
+* Join an existing organization If I'm invited
+
+Zones::
+* List of zones I have access to
+* See details of zone I have access to
+
+== Global Resources
+
+These are the resources which are global to the {product} instance.
+They are available in the global scope of the Kubernetes API server.
+
+=== `Organization` (Virtual)
+
+Organizations are the container for all resources which belong to an organization.
+It is represented by a Kubernetes Namespace with specific labels and annotations.
+This allows to use standard Kubernetes RBAC rules for organization-specific resources.
+
+To circumvent the current limitation of Kubernetes to show filtered lists of cluster-scoped resources based on access rights, the virtual object `Organization` represents an organization.
+
+The dynamic filter works through the list of namespaces which represent an organization and checks if the subject has access to the namespace.
+
+[source,yaml]
+----
+apiVersion: appuio.io/v1
+kind: Organization
+metadata:
+  name: acme-corp
+spec:
+  displayName: Acme Corp.
+----
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: org-acme-corp <1>
+  labels:
+    appuio.io/resource-type: organization <2>
+    organization.appuio.io/displayName: Acme Corp. <3>
+----
+<1> Prefixed (`org`) name to circumvent possible name collision (e.g. if a company is called "kube-system")
+<2> Identify resource type, used by the API server to filter for namespace representing organizations
+<3> Reflected in the organization object as `spec.displayName`
+
+Working with organization-resources always happens via the `Organization` object, it's not meant to directly CRUD namespaces which represent an organization.
+
+==== RBAC rules
+
+These are `ClusterRole` which can be bound to a subject by either a `ClusterRoleBinding` or a namespaced `RoleBinding`:
+
+`org-creator`:: Allow for creating organizations - on the `Organization` object (ClusterRoleBinding)
+`org-view`:: Namespace scope RoleBinding to allow view access to an organization
+`org-admin`:: Namespace scope RoleBinding to allow admin access to an organization (update, delete)
+
+=== `Invitation` and `InvitationAccept` (CRD)
+
+An invitation for a user to join an organization.
+
+Definition of being part of an organization:
+
+* At least `org-view` role in organization NS
+* Member of organization group in Keycloak
+
+Process:
+
+. An `Invitation` object is created in the organization namespace for which the invitation is meant for
++
+[source,yaml]
+----
+apiVersion: appuio.io/v1
+kind: Invitation
+metadata:
+  name: invite-1
+  namespace: org-acme-corp
+spec:
+  invitationcode: ju1aefaeKooz <1>
+  role: org-viewer <2>
+----
+<1> Generated invitation code
+<2> Role which this invitation assigns
```

Comment:
Who or what creates this invitation? Is it the org admin? Would we have to enforce rules like "min password length = 30 chars, uppercase, numbers etc"?
Regardless of that, with sharing some sort of code or token we would open up guessing/brute force attack surface.

Also, any other member of this org could share the code with someone else, that someone else could then hijack the invitation. could be even just an accidental leak from the org admin itself.